### PR TITLE
Fix email logic resolution on getTransport

### PR DESCRIPTION
**Pull request by StorytellerCZ**
In response to #11709 I'm adding more tests during which I have found one more edge case to account for in a logic and adding back a resolution for a logic error that I have fixed some while back, but can't seem to find in the code. Once I can replicate #11709 then I will create a test for it as well.

### CODE REVIEW

File: packages/email/email.js [hunk 1]
```diff
@@ -228,8 +240,9 @@ Email.send = function (options) {
     customTransport({ packageSettings, ...options });
     return;
   }
-  if (Meteor.isProduction || process.env.MAIL_URL || Meteor.settings.packages?.email) {
+  if (Meteor.isProduction && process.env.MAIL_URL || Meteor.settings.packages?.email) {
```
Comment (denihs) [inline packages/email/email.js:243]: I think you want to test here if it's production AND if there is an email URL, correct? If yes, you're missing some parenthesis:

```
if (Meteor.isProduction && (process.env.MAIL_URL || Meteor.settings.packages?.email)) {
```

&& has precedence over ||

**Review by denihs — CHANGES_REQUESTED** [review]

**Comment by denihs** [comment]
@StorytellerCZ Were you able to reproduce the issue?

File: packages/email/email.js (hunk 1 above)
Comment (StorytellerCZ) [reply packages/email/email.js:243]: That is something we need to decide. Both the fixed and your proposed conditions are valid options.

Comment (denihs) [reply packages/email/email.js:243]: So you're saying that if it's a development or staging environment but the `Meteor.settings.packages.email` is set, this code should run?

I don't think both conditions are valid because one says that this code should run in production if an email URL is set, as the other says that you can use settings to overwrite a behavior that should occur just in production.

Can you tell what the purpose of this code is? 

I'm not sure which one is correct. I will try to understand this code to see what it's supposed to happen there.

**Comment by denihs** [comment]
@StorytellerCZ, we analyzed and concluded the logic you changed was supposed to be like that, but some improvements should be made. So we did them [here](https://github.com/meteor/meteor/pull/11891). As soon as the next beta is out, I will let the people on [this](https://github.com/meteor/meteor/issues/11709) issue know, so they test out.
